src/main/java/core/basesyntax/operation/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        int fruitsInStorage = Storage.storage.get(fruitTransaction.getFruit());
+        int updateQuantity = fruitsInStorage + fruitTransaction.getQuantity();
+        Storage.storage.put(fruitTransaction.getFruit(), updateQuantity);
+    }
```

Comment:
You should check for the presence of the fruit in the storage before getting its quantity. If the fruit is not present, the `get` method will return `null`, which can lead to a `NullPointerException` when unboxing. Consider using `getOrDefault` method or check if the fruit is present in the storage before attempting to access its quantity.